File: app/src/main/java/org/deiverbum/app/data/entity/LiturgyEntity.java
package org.deiverbum.app.data.entity;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2022.2
 */

@Entity(tableName = "liturgy",
        indices = {@Index(value = {"timeFK","typeFK", "week", "day"}, unique = true)},

        foreignKeys =
        {
            @ForeignKey(
                    entity = LiturgyTimeEntity.class,
                    parentColumns = "timeID",
                    childColumns = "timeFK",
                    onDelete = ForeignKey.CASCADE,
                    onUpdate = ForeignKey.CASCADE),
            @ForeignKey(
                    entity = LiturgyColorEntity.class,
                    parentColumns = "colorID",
                    childColumns = "colorFK",
                    onDelete = ForeignKey.CASCADE,
                    onUpdate = ForeignKey.CASCADE)
        }
)
public class LiturgyEntity {

    @NonNull
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "liturgyID")
    public Integer liturgiaId=0;

    @NonNull
    @ColumnInfo(name = "timeFK")
    public Integer tiempoFK=0;

    @NonNull
    @ColumnInfo(name = "typeFK")
    public Integer tipoFK=0;

    @NonNull
    @ColumnInfo(name = "week")
    public Integer semana=0;

    @NonNull
    @ColumnInfo(name = "day")
    public Integer dia=0;

    @NonNull
    @ColumnInfo(name = "colorFK", index = true)
    public Integer colorFK=0;

    @NonNull
    @ColumnInfo(name = "name")
    public String nombre="";

    @NonNull
    public Integer getLiturgiaId() {
        return liturgiaId;
    }

    @SuppressWarnings("unused")
    @NonNull
    public Integer getTiempoFK() {
        return tiempoFK;
    }

    @SuppressWarnings("unused")
    @NonNull
    public Integer getTipoFK() {
        return tipoFK;
    }

    @NonNull
    public Integer getSemana() {
        return semana;
    }

    @NonNull
    public Integer getDia() {
        return dia;
    }

    @NonNull
    public Integer getColorFK() {
        return colorFK;
    }

    @NonNull
    public String getNombre() {
        return nombre;
    }


}

File: app/src/main/java/org/deiverbum/app/data/entity/LiturgyEntity.kt
package org.deiverbum.app.data.entity

import androidx.room.*
import androidx.room.ForeignKey.Companion.CASCADE

/**
 * Entidad para la tabla **`liturgy`** de la base de datos, que se ocupa de gestionar la Liturgia.
 *
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
@Entity(
    tableName = "liturgy",
    indices = [Index(value = ["timeFK", "typeFK", "week", "day"], unique = true)],
    foreignKeys = [ForeignKey(
        entity = LiturgyTimeEntity::class,
        parentColumns = arrayOf("timeID"),
        childColumns = arrayOf("timeFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    ), ForeignKey(
        entity = LiturgyColorEntity::class,
        parentColumns = arrayOf("colorID"),
        childColumns = arrayOf("colorFK"),
        onDelete = CASCADE,
        onUpdate = CASCADE
    )]
)
data class LiturgyEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "liturgyID")
    val liturgiaId: Int,

    @ColumnInfo(name = "timeFK")
    val tiempoFK: Int,

    @ColumnInfo(name = "typeFK")
    val tipoFK: Int,

    @ColumnInfo(name = "week")
    var semana: Int,

    @ColumnInfo(name = "day")
    var dia: Int,

    @ColumnInfo(name = "colorFK", index = true)
    var colorFK: Int,

    @ColumnInfo(name = "name")
    var nombre: String
)
